daemon/api_snaps.go

```diff
@@ -684,18 +685,19 @@ func snapEnforceValidationSets(inst *snapInstruction, st *state.State) (*snapIns
 		return nil, err
 	}
 
-	var validationErr *snapasserts.ValidationSetsValidationError
+	var tss []*state.TaskSet
+	var affected []string
 	err = assertstateTryEnforceValidationSets(st, inst.ValidationSets, inst.userID, snaps, ignoreValidationSnaps)
```

Comment:
this could return the keys of the new vs as well instead, for example?

Reply:
I opted to keep `TryEnforceValidationSets` out of it because it seemed more confusing. `Try...` could only return the extra set sometimes (an error might occur before getting them) so its interface becomes "we might or might not return the extra validation sets, depending on the error". Adding that data to the error solved that issue. A less concise but more consistent alternative would be for `Try...` to inspect the error and return `ValidationSetsValidationError` errors wrapped with the extra field.